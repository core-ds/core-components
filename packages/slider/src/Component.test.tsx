import React from 'react';
import { render } from '@testing-library/react';

import { Slider } from './index';

describe('Slider', () => {
    it('should match snapshot', () => {
        const { container } = render(<Slider step={1} />);
        expect(container).toMatchSnapshot();
    });

    it('should set `data-test-id` attribute', () => {
        const dataTestId = 'test-id';
        const { getByTestId } = render(<Slider dataTestId={dataTestId} />);

        expect(getByTestId(dataTestId)).toBeTruthy();
    });

    it('should set `className` class to root', () => {
        const className = 'test-class';
        const { container } = render(<Slider className={className} />);

        expect(container.firstElementChild).toHaveClass(className);
    });

    it('should set min, max correctly', () => {
        const dataTestId = 'test-id';
        const min = 5;
        const max = 50;
        const { getByRole } = render(<Slider min={min} max={max} dataTestId={dataTestId} />);
        const slider = getByRole('slider') as HTMLInputElement;

        expect(slider).toHaveAttribute('aria-valueMin', min.toFixed(1));
        expect(slider).toHaveAttribute('aria-valueMax', max.toFixed(1));
    });

    it('should set value correctly', () => {
        const dataTestId = 'test-id';
        const value = 10;
        const min = 5;
        const max = 50;
        const step = 5;
        const { getByRole } = render(
            <Slider value={value} min={min} max={max} step={step} dataTestId={dataTestId} />,
        );

        expect(getByRole('slider')).toHaveAttribute('aria-valuenow', value.toFixed(1));
    });

    it('should set out of bounds value in range [min, max]', () => {
        const dataTestId = 'test-id';
        const min = 5;
        const max = 55;
        const valueBelowMinimum = 4;
        const valueAboveMaximum = 56;

        const { getByRole, rerender } = render(
            <Slider value={valueBelowMinimum} min={min} max={max} dataTestId={dataTestId} />,
        );

        expect(getByRole('slider')).toHaveAttribute('aria-valuenow', min.toFixed(1));

        rerender(<Slider value={valueAboveMaximum} min={min} max={max} dataTestId={dataTestId} />);

        expect(getByRole('slider')).toHaveAttribute('aria-valuenow', max.toFixed(1));
    });

    it('should unmount without errors', () => {
        const { unmount } = render(<Slider />);

        expect(unmount).not.toThrowError();
    });
});
