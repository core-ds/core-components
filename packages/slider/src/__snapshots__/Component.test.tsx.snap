// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Slider should match snapshot 1`] = `
<div>
  <div
    class="component s noUi-target noUi-ltr noUi-horizontal noUi-txt-dir-ltr noUi-state-tap"
  >
    <div
      class="noUi-base"
    >
      <div
        class="noUi-connects"
      >
        <div
          class="noUi-connect"
          style="transform: translate(0%, 0) scale(0, 1);"
        />
      </div>
      <div
        class="noUi-origin"
        style="transform: translate(-100%, 0); z-index: 4;"
      >
        <div
          aria-orientation="horizontal"
          aria-valuemax="100.0"
          aria-valuemin="0.0"
          aria-valuenow="0.0"
          aria-valuetext="0.00"
          class="noUi-handle noUi-handle-lower"
          data-handle="0"
          role="slider"
          tabindex="0"
        >
          <div
            class="noUi-touch-area"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
